integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/EnvironmentChildTest.java

```diff
@@ -0,0 +1,180 @@
+
+import static java.util.Objects.isNull;
+import static org.mockito.ArgumentMatchers.any;
+
+import java.util.Collection;
+
+import javax.inject.Inject;
+import javax.ws.rs.BadRequestException;
+
+import org.mockito.Mockito;
+import org.testng.annotations.Test;
+
+import com.sequenceiq.environment.api.v1.environment.model.response.EnvironmentStatus;
+import com.sequenceiq.environment.api.v1.environment.model.response.SimpleEnvironmentResponse;
+import com.sequenceiq.it.cloudbreak.EnvironmentClient;
+import com.sequenceiq.it.cloudbreak.client.EnvironmentTestClient;
+import com.sequenceiq.it.cloudbreak.context.Description;
+import com.sequenceiq.it.cloudbreak.context.MockedTestContext;
+import com.sequenceiq.it.cloudbreak.context.RunningParameter;
+import com.sequenceiq.it.cloudbreak.context.TestContext;
+import com.sequenceiq.it.cloudbreak.dto.environment.EnvironmentTestDto;
+import com.sequenceiq.it.cloudbreak.exception.TestFailException;
+import com.sequenceiq.it.cloudbreak.mock.freeipa.AbstractFreeIpaResponse;
+import com.sequenceiq.it.cloudbreak.testcase.AbstractIntegrationTest;
+
+public class EnvironmentChildTest extends AbstractIntegrationTest {
+
+    private static final String PARENT_ENVIRONMENT = "parent";
+
+    private static final String CHILD_ENVIRONMENT = "child";
+
+    @Inject
+    private EnvironmentTestClient environmentTestClient;
+
+    @Override
+    protected void setupTest(TestContext testContext) {
+        createDefaultUser(testContext);
+        initializeDefaultBlueprints(testContext);
+        createDefaultCredential(testContext);
+        createDefaultEnvironmentWithNetwork(testContext);
+        createDefaultFreeIPA(testContext);
+    }
+
+    @Test(dataProvider = TEST_CONTEXT_WITH_MOCK)
+    @Description(
+            given = "there is an available parent environment",
+            when = "valid create child environment request is sent",
+            then = "environment should be created and parent environment should be referenced in the child environment")
+    public void testCreateChildEnvironment(MockedTestContext testContext) {
+        AbstractFreeIpaResponse<?> dnsZoneAddResponse = Mockito.mock(AbstractFreeIpaResponse.class);
+        getFreeIpaRouteHandler().updateResponse("dnszone_add", dnsZoneAddResponse);
+
+        testContext
+                .given(EnvironmentTestDto.class)
+                .when(environmentTestClient.list())
+                .then(this::checkEnvIsListedByNameAndParentName)
+                .given(CHILD_ENVIRONMENT, EnvironmentTestDto.class)
+                    .withParentEnvironment()
+                .when(environmentTestClient.create())
+                .await(EnvironmentStatus.AVAILABLE)
+                .when(environmentTestClient.list())
+                .then(this::checkEnvIsListedByNameAndParentName)
+                .then((testContext1, testDto, client) -> {
+                    Mockito.verify(dnsZoneAddResponse).handle(any(), any());
```

Comment:
Although it is working, this is not the way what we use to follow in our mock tests. We are using a httpmock (for FreeIPA as well), and we are use to seek the http mock calls for validation. You can see examples in DistroXClusterCreationTest